### DISCUSSION TITLE
Secure Tags for Fw and parent id

### DISCUSSION THREAD

**Discussion by Sekyrovat** [GoogleCloudPlatform/cloud-foundation-fabric#2345] [General]
Hi there folks,

It seems that as of now, the config that is present when trying to use module [organization](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/organization) you are forced to create the secure tags with the Parent being the Organization only.

As per what is noted in the [docs](https://cloud.google.com/firewall/docs/tags-firewalls-overview#differences) it seems that it should be possible to define the tag with a parent being the Org or a project. I believe that this spans from the condition being checked in the organization_id variable [ref](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/organization/variables.tf#L92).

Would it make sense to define a new variable for using the project as parent rather than only allowing only using organizations?

**Comment by ludoo** [comment]
You can already define tags at the project level, just use the project module. :)

**Comment by Sekyrovat** [comment]
Ahhhhh there we go, knew there'd be a gocha odd the only ref I found to tags was in the Org module ;.; xD but tyty @ludoo !

update: Seems the project module didn't quite have the expected outcome :/

**Reply by ludoo** [reply]
Yes, typically if you need something at the project level that lives in the project module. We take a resource-centric view of things. :) Let us know if you have more questions!

**Reply by Sekyrovat** [reply]
So currently using it (or trying to) seems the way the outputs are done, and there are some other requirements when using the project module, and it is not quite as seamless as I hoped for but will keep trying for now.

**Reply by Sekyrovat** [reply]
I think I see the problem with this module and why I avoided it the first time around. Using this module you don't get the output to leverage the tags (keys and values) in terraform in other modules, unlike the Organization module.

Piped the whole module result as an output and tagkeys and tagvalues were empty, with no note of network_tags

**Reply by ludoo** [reply]
Network and resource tags are different, I thought you were going to use resource tags. Can you open an issue mentioning this? I can send a fix tomorrow.

**Reply by Sekyrovat** [reply]
Filed the [issue tracker](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/issues/2348), would love to collaborate with you on it @ludoo ! Lemme know if you have time on Friday to discuss :3

**Reply by ludoo** [reply]
Can you go through the Contributing guide in the root of the repo? Then I'll point you to the bits that need to be carried over to the project module. If you are up for a PR, that would be great.

**Reply by Sekyrovat** [reply]
Sounds like a plan, doe tbh may take me a bit (full disclosure xd)

**Reply by ludoo** [reply]
no worries, take your time: you're the first to come up with this use case; on the other hand if you are in a rush and prefer us to do it, no problems at all

**Comment by ludoo** [comment]
closing this as fix is in #2350
